### PR TITLE
Feat add long docs classification tasks

### PR DESCRIPTION
**Pull request by guenthermi**
This PR adds to classification tasks for evaluating models on long texts.

### CODE REVIEW

File: mteb/tasks/Classification/LongDocumentClassification.py [hunk 1]
```diff
@@ -0,0 +1,35 @@
...
+
+class PatentClassification(AbsTaskClassification):
+    @property
+    def description(self):
+        return {
+            "name": "PatentClassification",
+            "hf_hub_name": "jinaai/small_patent",
+            "description": "Patent classification evaluation based on the test set of the big patent dataset",
+            "reference": "https://dblp.org/rec/journals/corr/abs-1906-03741.bib",
+            "category": "p2p",
```
Comment (jupyterjazz) [inline mteb/tasks/Classification/LongDocumentClassification.py:12]: what does p2p mean?

Comment (guenthermi) [reply mteb/tasks/Classification/LongDocumentClassification.py:12]: It is one of the task categories of MTEB. basically, it means that classification is not applied on small sentences but  rather on paragraphs.

Comment (jupyterjazz) [reply mteb/tasks/Classification/LongDocumentClassification.py:12]: ok cool

**Review by jupyterjazz — APPROVED** [review]

**Review by gmastrapas — APPROVED** [review]

**Review by bwanglzu — APPROVED** [review]
LGTM!

**Pull request merged**